api/login_api.py

```diff
@@ -34,12 +32,14 @@ def do_post(self):
     message = "Unable to Authenticate"
 
     try:
-      unused_idinfo = id_token.verify_oauth2_token(
+      idinfo = id_token.verify_oauth2_token(
           token, requests.Request(),
           settings.GOOGLE_SIGN_IN_CLIENT_ID)
-      # userid = idinfo['sub']
-      # email = idinfo['email']
-      session["id_token"] = token
+      user_info = {
+          'email': idinfo['email'],
+          }
+      signature = xsrf.generate_token(str(user_info))
```

Comment:
We should also encode `user_info`. User shouldn't able to view what the value for `signed_user_info` looks like

Reply:
Flask encrypts the entire session using app.secret_key (set in basehandlers.py).

That raises the question, why do we need to sign the user_info if it is already encrypted using some kind of secret?  Encryption + signing seems wasteful.  I think it is wise to sign the user_info separately so that we can implement key rotation on the signing key.  I don't see a way to implement key rotation for Flask's app.secret_key.

Reply:
I see. Well I can speak for the [securecookie library](https://github.com/gorilla/securecookie#securecookie) we use in wpt.fyi. This library `hashKey`, `blockKey` where `hashkey` uses for cookie singing and `blockkey` encrypts the value itself. I cannot find the documentation for the flask cookie encryption, so my question is does it do both encryption and signature? If so, I am open not to have a key rotation

Reply:
I only found it by looking at the flask source code.  I don't see anything there about key rotation.  So, I think we will need to do it for `user_info` but let's see what they suggest in the security review rather than adding complexity now.

Reply:
Yea lets do that